community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/pipes/ValueHashJoinPipe.scala

```diff
@@ -49,7 +50,8 @@ case class ValueHashJoinPipe(lhsExpression: Expression, rhsExpression: Expressio
     val result = for {context: ExecutionContext <- rhsIterator
                       joinKey = rhsExpression(context) if joinKey != null}
       yield {
-        val seq = table.getOrElse(joinKey, mutable.MutableList.empty)
+        val equiKey = Equivalent(joinKey)
```

Comment:
Maybe the code base is naming things wrongly - this PR does the same thing that `Equals` does: https://github.com/neo4j/neo4j/blob/3.1/community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/commands/predicates/ComparablePredicate.scala#L71